File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q12-largest-divisible-subset.js
var abc = function(nums) {
    if (nums.length == 0) {
        return [];
    }
    nums.sort(function(a, b) {
        return a - b;
    });
    var dp = new Array(nums.length);
    dp[0] = [nums[0]];
    var max = dp[0];
    for (var i = 1; i < nums.length; i++) {
        var maxIndex = -1;
        for (var j = i - 1; j >= 0; j--) {
            if (nums[i] % nums[j] == 0) {
                if (maxIndex == -1 || dp[j].length > dp[maxIndex].length) {
                    maxIndex = j;
                }
            }
        }
        if (maxIndex == -1) {
            dp[i] = [nums[i]];
        } else {
            dp[i] = dp[maxIndex].concat([nums[i]]);
        }
        if (dp[i].length > max.length) {
            max = dp[i];
        }
    }
    return max;
}

File: ELEC881_Copilot/src/Function Name Masked/Python/q12-largest-divisible-subset.py
class abc(object):
    def bcd(self, nums):
        if not nums:
            return []
        nums.sort()
        dp = [1] * len(nums)
        parent = [-1] * len(nums)
        max_index = 0
        max_len = 1
        for i in range(len(nums)):
            for j in range(i):
                if nums[i] % nums[j] == 0 and dp[i] < dp[j] + 1:
                    dp[i] = dp[j] + 1
                    parent[i] = j
                    if dp[i] > max_len:
                        max_len = dp[i]
                        max_index = i
        res = []
        while max_index != -1:
            res.append(nums[max_index])
            max_index = parent[max_index]
        return res[::-1]